tripal/src/Entity/TripalEntity.php

```diff
@@ -320,37 +393,84 @@ public function getFieldValues() {
   }
 
   /**
-   * Replaces tokens in URL Aliases and Titles.
+   * Flattens the field values to be suitable for use as values
+   * for token replacement. Values with cardinality > 1 are
+   * excluded.
    *
-   * @param string $string
-   *   The string to replace.
-   * @param array $cache
+   * @param array $field_values
+   *   Values nested array from $this->getFieldValues()
+   * @return array
+   *   Associative array of key => value pairs
    */
-  protected function replaceTokens($string, $bundle) {
+  protected function processFieldValues(array $field_values): array {
+    $processed_values = [];
+    foreach ($field_values as $field => $value_array) {
+      // Token replacement only supports single-value fields,
+      // therefore, only add if the items array has a single value.
+      if (count($value_array) == 1) {
+        if (array_key_exists('value', $value_array[0])) {
+          $processed_values[$field] = $value_array[0]['value'];
+        }
```

Comment:
I believe the old code supported tokens specific to a given field property. For example, one a gene content type there would be a token for the organism.genus... or at least if it didn't just yet, it was designed to be able to by adding all the properies of a field to the token values list. This was done with the following which has since been removed.
```
        $props = $item->getProperties();
        foreach ($props as $prop) {
          $token_parser->addFieldValue($field_name, $prop->getName(), $prop->getValue());
        }
```
I would like to see this added back. To keep with your current flat array approach, you could keep the `$processed_values[$field] = $value_array[0]['value'];` approach for each field but then also add in the properties like this:
```
        $props = $item->getProperties();
        foreach ($props as $prop) {
          $token = $field . "." . $prop->getName();
          $processed_values[$token] = $prop->getValue();
        }
```
This way your current functionality remains unchanged but the property values are still in the list of token values for when we support that in the future.

Reply:
I'm not sure I understand fully, currently we do this in `getFieldValues()`
```
            $prop_name = $prop->getName();
            $prop_value = $prop->getValue();
            $values[$field_name][$delta][$prop_name] = $prop_value;
```
so I think all of the values should be there already.
Then these get flattened in `processFieldValues()`
Here I created an organism with all the fields with values
![processFieldValues 1](https://github.com/user-attachments/assets/03a38871-e306-4ec0-95df-68a0c5dc6b68)

Here just genus and species.
![processFieldValues 2](https://github.com/user-attachments/assets/99442382-80f4-45ba-956f-fa248ba55055)

So the dbxref and publication do not show up if they don't have a value, is that what you mean?

Reply:
Moved into Issue #2065 since it is beyond the scope of this PR.